src/dmd/link.d

```diff
@@ -562,59 +562,148 @@ public int runLINK()
             argv.push("-Xlinker");
             argv.push("--gc-sections");
         }
+
+        /// Dynamic array of strings to temporarily hold accumulated,
+        /// reordered linker args
+        Strings linkerArgs;
+
+        /**
+        Checks if a C string starts with a given string
+        Params:
+            p = the C string to check
+            needle = the string to look for
+        Returns
+            `true` if `p` starts with `needle`
+        */
+        static bool startsWith(const(char)* p, string needle)
+        {
```

Comment:
Put null check here instead of at the caller.

Reply:
Addressed in https://github.com/dlang/dmd/pull/10341  Please take another look.